packages/x-components/src/x-modules/history-queries/store/actions/update-history-queries-with-search-response.action.ts

```diff
@@ -0,0 +1,29 @@
+import { HistoryQuery } from '@empathyco/x-types';
+import { HistoryQueriesXStoreModule } from '../types';
+
+/**
+ * Default implementation for the
+ * {@link HistoryQueriesActions.updateHistoryQueriesWithSearchResponse} action.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @param searchResponse - The search response to update history queries with.
+ *
+ * @returns A `void` promise that resolves when the history query finishes updating.
+ *
+ * @public
+ */
+// eslint-disable-next-line max-len
+export const updateHistoryQueriesWithSearchResponse: HistoryQueriesXStoreModule['actions']['updateHistoryQueriesWithSearchResponse'] =
+  ({ state, dispatch }, searchResponse) => {
+    if (searchResponse.status === 'success') {
```

Comment:
The main problem with this approach is here, and it is about when to update the total results count of a history query.

Right now every successfuly response updates it, but that means it will be updated when sorting, loading a new page of the same query, and even filtering which decreases the count and that is irrelevant for the history queries.